packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -341,10 +350,12 @@ See documentation: ${createDocumentationLink({
     this.future = future;
     this.insightsClient = insightsClient;
     this.indexName = indexName;
+    this.compositionID = compositionID;
     this.helper = null;
     this.mainHelper = null;
     this.mainIndex = index({
-      indexName,
+      // we use an index widget to render compositions
+      indexName: this.compositionID || this.indexName,
```

Comment:
```suggestion
      indexName,
      compositionID,
```

(and accept and read the compositionID in index widget, to put it in the state in mergeSearchParameters)